python/dgl/graphbolt/minibatch.py

```diff
@@ -433,25 +433,63 @@ def node_pairs_with_labels(self):
         else:
             return None
 
-    def to(self, device: torch.device) -> None:  # pylint: disable=invalid-name
+    def to(self, device, extra_attrs=None):  # pylint: disable=invalid-name
         """Copy `MiniBatch` to the specified device using reflection."""
 
         def _to(x, device):
             return x.to(device) if hasattr(x, "to") else x
 
-        for attr in dir(self):
-            # Only copy member variables.
-            if not callable(getattr(self, attr)) and not attr.startswith("__"):
-                try:
-                    setattr(
-                        self,
-                        attr,
-                        recursive_apply(
-                            getattr(self, attr), lambda x: _to(x, device)
-                        ),
-                    )
-                except AttributeError:
-                    continue
+        def apply_to(x, device):
+            return recursive_apply(x, lambda x: _to(x, device))
+
+        if self.labels is not None and self.seed_nodes is not None:
+            # Node classification.
+            self.labels = apply_to(self.labels, device)
```

Comment:
Is it possible to obtain the target attributes to be copied first and call `apply_to()` in a for loop? Current implementation looks a bit tedious.